storage/series/series.go

```diff
@@ -440,6 +442,31 @@ func (s *dbSeries) mergeBlock(
 	return nil
 }
 
+func (s *dbSeries) Update(blk block.DatabaseBlock) error {
+	s.Lock()
+	defer s.Unlock()
+
+	// ensure series is bootstraped before allowing any updates
+	if s.bs != bootstrapped {
+		return errSeriesNotBootstrapped
+	}
+
+	// we only accept updates to non-buffered data
+	// ensure buffer is past provided time
+	min, _ := s.buffer.MinMax()
+	if !blk.StartTime().Before(min) {
```

Comment:
Yeah this looks good, it won't try to repair any block that could still receive writes.  The past 2 hour block can still be written into for the entire duration of buffer past past when its block end time is (by default up to 10mins after).